pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -334,7 +334,7 @@ func (c *ReplicaCalculator) GetExternalMetricReplicas(currentReplicas int32, tar
 // GetExternalPerPodMetricReplicas calculates the desired replica count based on a
 // target metric value per pod (as a milli-value) for the external metric in the
 // given namespace, and the current replica count.
-func (c *ReplicaCalculator) GetExternalPerPodMetricReplicas(currentReplicas int32, targetUtilizationPerPod int64, metricName, namespace string, metricSelector *metav1.LabelSelector) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
+func (c *ReplicaCalculator) GetExternalPerPodMetricReplicas(statusReplicas int32, targetUtilizationPerPod int64, metricName, namespace string, metricSelector *metav1.LabelSelector) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
```

Comment:
Why "statusReplica". Is increasing number of pods during rolling upgrade increasing external metric? Additional pods  added during upgrade shouldn't serve as if they are healthy then old pods are deleted.

Reply:
Incidentally, it doesn't make a difference whether we use status or spec replicas here.  It's technically a bug to use status replicas, but I left it as-is in order to fix one bug at a time.  https://github.com/kubernetes/kubernetes/issues/79274 tracks fixing the bug to use spec replicas.

Reply:
Also, it's super confusing that we pass in spec replicas at all.  We only use that value in order to signal a decision to not scale by returning the current desired replica count.  Better would be to return a boolean.  https://github.com/kubernetes/kubernetes/issues/79275 tracks that improvement.